src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -360,6 +360,11 @@ uct_dc_mlx5_ep_basic_init(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
                                           ep->dci);
         }
     } else if (uct_dc_mlx5_iface_is_hw_dcs(iface)) {
+        if (ucs_array_is_empty(&iface->tx.dcis)) {
+            uct_dc_mlx5_iface_resize_and_fill_dcis(iface, 1);
+            uct_dc_mlx5_dci_pool_init_dci(iface, uct_dc_mlx5_ep_pool_index(ep),
```

Comment:
maybe move `uct_dc_mlx5_iface_resize_and_fill_dcis` into `uct_dc_mlx5_dci_pool_init_dci`?

Reply:
in random policy we may need to resize/fill DCIs up to certain index, but not initialize them (yet)